### PR TITLE
RFR: Added CSV upload functionality (needs backend fix)

### PR DESCRIPTION
**Pull request by josh-byster**
## Status: 

🚀 

## Description
A few sentences describing the overall goals of the pull request's commits.
Adds the frontend functionality to upload a CSV. This will POST the data to the backend properly, however there is an issue with the backend not parsing the data (seems like there's something going on with invoking `next` on the CSV stream). Hence, it displays that the book has been uploaded successfully but there is no change in the db.
 
Related issues: #158
Related PRs: #<number>

## Todos

Backend fix

## Screenshots

image

### CODE REVIEW

File: frontend/src/pages/admin/AdminCSVUploadPage.js [hunk 1]
```diff
@@ -21,7 +23,19 @@ class AdminCSVUploadPage extends Component {
...
 
-  handleSubmit = () => {};
+  handleSubmit = async (event, formState) => {
+    event.preventDefault();
+    const { message, success } = await bookFromCSV(this.state.file);
+    if (success) {
+      this.setState({ success: true });
+    } else {
+      this.setState(state => ({
+        errors: [{ message: message, key: state.numSubmits }],
```
Comment (Davidw1339) [inline frontend/src/pages/admin/AdminCSVUploadPage.js]: Just wondering why errors is an array here

**Comment by josh-byster** [comment]
Thanks David!!

File: frontend/src/pages/admin/AdminCSVUploadPage.js (hunk 1 above)
Comment (josh-byster) [reply frontend/src/pages/admin/AdminCSVUploadPage.js]: Good question - I had it there so we could support if BE returned multiple errors, but if that's not the case then we are free to make it a single object.

Comment (Davidw1339) [reply frontend/src/pages/admin/AdminCSVUploadPage.js]: Ahhh that makes sense, unfortunately backend just returns the first error it encounters while trying to load in the books so we could probably just make it a single error

**Pull request merged**